contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3276,12 +3323,11 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 						}
 						PG_FINALLY();
 						{
-							SetCurrentRoleId(prev_current_user, true);
+							SetUserIdAndSecContext(save_userid, save_sec_context);
```

Comment:
Right, transaction abort/cleanup handles the clean up of this local CurrentUserId change. But I think this might be more safe, since we really do not want to leak this state.